graphx/src/main/scala/org/apache/spark/graphx/Pregel.scala

```diff
@@ -155,6 +169,8 @@ object Pregel extends Logging {
       i += 1
     }
     messages.unpersist(blocking = false)
+    graphCheckpointer.deleteAllCheckpoints()
+    messageCheckpointer.deleteAllCheckpoints()
```

Comment:
shouldn't this be inside a finally clause to make sure checkpoint data is cleaned up even in error cases?